File: HASHING/unordered_map.java
import java.util.HashMap;

public class unordered_map{
    public static void main(String[] args) {
        HashMap<String, Integer> hashTable = new HashMap<>();

        // Insert key-value pairs
        hashTable.put("apple", 10);
        hashTable.put("banana", 20);
        hashTable.put("orange", 30);

        // Access values
        System.out.println("apple: " + hashTable.get("apple"));
        System.out.println("banana: " + hashTable.get("banana"));

        // Search for a key
        String key = "orange";
        if (hashTable.containsKey(key)) {
            System.out.println(key + " found with value: " + hashTable.get(key));
        } else {
            System.out.println(key + " not found!");
        }
    }
}

File: HASHING/unordered_map.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    unordered_map<string, int> hashTable;

    // Insert key-value pairs
    hashTable["apple"] = 10;
    hashTable["banana"] = 20;
    hashTable["orange"] = 30;

    // Access values
    cout << "apple: " << hashTable["apple"] << endl;
    cout << "banana: " << hashTable["banana"] << endl;

    // Search for a key
    string key = "orange";
    if (hashTable.find(key) != hashTable.end()) {
        cout << key << " found with value: " << hashTable[key] << endl;
    } else {
        cout << key << " not found!" << endl;
    }

    return 0;
}
